pkg/sql/schemaexpr/expr.go

```diff
@@ -131,3 +135,84 @@ func ExtractColumnIDs(
 
 	return colIDs, err
 }
+
+// NameResolver is used to replace unresolved names in expressions with
+// IndexedVars.
+type NameResolver struct {
+	evalCtx    *tree.EvalContext
+	tableID    sqlbase.ID
+	cols       []sqlbase.ColumnDescriptor
```

Comment:
We want to generally carry slices of `*ColumnDescriptor`, since these are getting expensive to copy around

Reply:
Done.